utils/swift_build_support/swift_build_support/products/llvm.py

```diff
@@ -291,18 +291,71 @@ def build(self, host_target):
         llvm_cmake_options.define('LLVM_INCLUDE_DOCS:BOOL', 'TRUE')
         llvm_cmake_options.define('LLVM_ENABLE_LTO:STRING', self.args.lto_type)
         llvm_cmake_options.define('COMPILER_RT_INTERCEPT_LIBDISPATCH', 'ON')
+        # Swift expects the old layout for the runtime directory
+        # updating this in tracked in #80180
+        llvm_cmake_options.define('LLVM_ENABLE_PER_TARGET_RUNTIME_DIR', 'OFF')
+        if host_target.startswith('linux'):
+            # This preserves the behaviour we had when using
+            # LLVM_BUILD_EXTERNAL COMPILER_RT --
+            # that is, having the linker not complaing if symbols used
+            # by TSan are undefined (namely the ones for Blocks Runtime)
+            # In the long term, we want to remove this and
+            # build Blocks Runtime before LLVM
+            llvm_cmake_options.define(
+                'SANITIZER_COMMON_LINK_FLAGS:STRING', '-Wl,-z,undefs')
+
+        builtins_runtimes_target_for_darwin = 'arm64-apple-darwin'
+        if system() == "Darwin":
+            llvm_cmake_options.define(
+                'LLVM_BUILTIN_TARGETS', builtins_runtimes_target_for_darwin)
+            llvm_cmake_options.define(
+                'LLVM_RUNTIME_TARGETS', builtins_runtimes_target_for_darwin)
+            llvm_cmake_options.define('RUNTIMES_BUILD_ALLOW_DARWIN', 'ON')
+            # Build all except rtsan
+            llvm_cmake_options.define(
+                f'RUNTIMES_{builtins_runtimes_target_for_darwin}_'
+                'COMPILER_RT_SANITIZERS_TO_BUILD',
+                'asan;dfsan;msan;hwasan;tsan;safestack;cfi;scudo_standalone;'
+                'ubsan_minimal;gwp_asan;nsan;asan_abi')
 
         if self.args.build_embedded_stdlib and system() == "Darwin":
             # Ask for Mach-O cross-compilation builtins (for Embedded Swift)
             llvm_cmake_options.define(
                 'COMPILER_RT_FORCE_BUILD_BAREMETAL_MACHO_BUILTINS_ARCHS:STRING',
                 'armv6 armv6m armv7 armv7m armv7em')
+            llvm_cmake_options.define(
+                f'BUILTINS_{builtins_runtimes_target_for_darwin}_'
+                'COMPILER_RT_FORCE_BUILD_BAREMETAL_MACHO_BUILTINS_ARCHS:'
+                'STRING', 'armv6 armv6m armv7 armv7m armv7em')
 
         llvm_enable_projects = ['clang']
+        llvm_enable_runtimes = []
 
         if self.args.build_compiler_rt and \
                 not self.is_cross_compile_target(host_target):
-            llvm_enable_projects.append('compiler-rt')
+            if self.args.llvm_build_compiler_rt_with_use_runtimes:
+                llvm_enable_runtimes.append('compiler-rt')
+                build_targets.append("runtimes")
```

Comment:
This is not needed, as `all` takes care of building the runtimes as well -- users of `--llvm_ninja_targets` will need to provide `runtimes`  target is needed

```suggestion
```

Making this change to #80253 to begin with, likely I will apply this here as well together with future feedback.